pageserver/src/tenant/layer_map/historic_layer_coverage.rs

```diff
@@ -417,13 +417,11 @@ impl<Value: Clone> BufferedHistoricLayerCoverage<Value> {
         }
     }
 
-    pub fn insert(&mut self, layer_key: LayerKey, value: Value) {
-        // layer duplicates are not allowed
-        assert!(!self.layers.contains_key(&layer_key));
-        self.buffer.insert(layer_key, Some(value));
+    pub fn contains(&self, layer_key: LayerKey) -> bool {
+        self.layers.contains_key(&layer_key)
     }
 
-    pub fn replace_existed(&mut self, layer_key: LayerKey, value: Value) {
```

Comment:
Unsure why can't I suggest here, but `replace_existed` should be `replace_existing` ... Though, the only place which uses it doesn't actually want to do that. It wants to `replace_buffered`, so perhaps this fn makes no sense, should just do the overwrite in `replace`.

Reply:
It was undo of my previous attempt